helm-ag.el

```diff
@@ -367,7 +367,8 @@ Default behaviour shows finish and result in mode-line."
   "Not documented, FILENAME."
   (let ((search-directory default-directory))
     (switch-to-buffer (get-buffer-create " *helm-ag persistent*"))
-    (setq default-directory search-directory)
+    (setq default-directory search-directory
+          buffer-read-only nil)
```

Comment:
I think you should probably wrap `buffer-ready-only` inside the let scope?

Reply:
I'm not sure I follow. Do you mean move it up like this:
```elisp
(let ((search-directory default-directory)
      (buffer-read-only nil))
  (switch-to-buffer (get-buffer-create " *helm-ag persistent*"))
  (setq default-directory search-directory)
```

If so, then I don't think that works: we want to set the buffer-local `buffer-read-only` variable in the buffer returned by `(get-buffer-create " *helm-ag persistent*")`. So it must be set *after* switching.

Reply:
@jcs090218 Does that make sense to you?